File: Mobile/CRMSystemMobile/Services/SkillService.cs
using System.Net.Http.Json;
using CRMSystemMobile.Extensions;
using Shared.Contracts.Skill;
using Shared.Filters;

namespace CRMSystemMobile.Services;

public class SkillService(HttpClient httpClient)
{
    public async Task<List<SkillResponse>?> GetWorkerSkills(int workerId)
    {
        var filter = new SkillFilter(
            WorkerIds: [workerId],
            SpecializationIds: [],
            SortBy: null,
            IsDescending: false
        );

        var (items, _) = await httpClient.GetPagedAsync<SkillResponse>("api/v1/skills", filter);
        return items;
    }

    public async Task<string?> CreateSkill(SkillRequest request)
    {
        try
        {
            var response = await httpClient.PostAsJsonAsync("api/v1/skills", request);
            if (response.IsSuccessStatusCode)
            {
                return null;
            }

            return await response.Content.ReadAsStringAsync();
        }
        catch (Exception ex)
        {
            return ex.Message;
        }
    }

    public async Task<string?> DeleteSkill(int id)
    {
        try
        {
            var response = await httpClient.DeleteAsync($"api/v1/skills/{id}");
            if (response.IsSuccessStatusCode)
            {
                return null;
            }

            return await response.Content.ReadAsStringAsync();
        }
        catch (Exception ex)
        {
            return ex.Message;
        }
    }
}

File: Web(New)/crm-frontend/src/api/services/skillService.ts
import { apiClient } from "../config.ts";
import type {
    SkillFilter,
    SkillRequest,
    SkillResponse
} from "../types/skill.ts";

export const skillService = {
    getPaged: async (filter: SkillFilter) => {
        return (await apiClient.get<SkillResponse[]>('/skills', { params: filter })).data;
    },

    getById: async (id: number) => {
        return (await apiClient.get<SkillResponse>(`/skills/${id}`)).data;
    },

    create: async (request: SkillRequest) => {
        return (await apiClient.post<SkillResponse>('/skills', request)).data;
    },

    delete: async (id: number) => {
        return (await apiClient.delete(`/skills/${id}`)).data;
    }
};
